commons/all/src/main/java/org/infinispan/commons/util/TypedProperties.java

```diff
@@ -54,20 +54,25 @@ public int getIntProperty(String key, int defaultValue) {
    }
 
    public int getIntProperty(String key, int defaultValue, boolean doStringReplace) {
-      String value = getProperty(key);
-      if (value == null) return defaultValue;
-      value = value.trim();
-      if (value.length() == 0) return defaultValue;
-
-      if (doStringReplace)
-         value = StringPropertyReplacer.replaceProperties(value);
-
-      try {
-         return Integer.parseInt(value);
-      }
-      catch (NumberFormatException nfe) {
-         log.unableToConvertStringPropertyToInt(value, defaultValue);
-         return defaultValue;
+      Object value = this.get(key);
+      if (value instanceof Integer) {
+         return (int) value;
+      } else {
+         String valueStr = getProperty(key);
```

Comment:
avoid invoking `getProperty(key)`. check if the `value` and use it.

Reply:
I think I can't remove the `getProperty` because it looks into the `map` and has a fallback in the `defaults`.

Reply:
oh. I wasn't aware of the `defaults`. 
In that case, your code needs to check `defaults`, If `value = this.get(key)` is null. (IMO)